spring-integration-redis/src/main/java/org/springframework/integration/redis/util/RedisLockRegistry.java

```diff
@@ -305,6 +338,32 @@ public boolean tryLock(long time, TimeUnit unit) throws InterruptedException {
 			return false;
 		}
 
+		private boolean subscribeLock() throws ExecutionException, InterruptedException {
+			return subscribeLock(-1L);
+		}
+
+		private boolean subscribeLock(long time)
+				throws ExecutionException, InterruptedException {
+			if (!obtainLock()) {
+				if (!RedisLockRegistry.this.redisMessageListenerContainer.isRunning()) {
+					RedisLockRegistry.this.redisMessageListenerContainer.start();
+				}
+				final Future future = RedisLockRegistry.this.unlockNotifyMessageListener.unlockWait(this.lockKey);
```

Comment:
I think the difference between Thread.sleep() and future.get() is
I think there is a difference between purpose and frequency.

If the purpose of Thread.sleep() was to simply repeat

future.get() doesn't need to check repeatedly until unlock() is done.
However, since ttl exists, I set the time. 
it must be able to wake up without unlocking

Usually we expect to use the default value (1min), so we don't need to retry in the meantime.
And if you perform unLock() faster than 100ms in another vm, you can acquire the lock faster than 100ms.

second, 
I think the use for unlink should be considered.
That part is not reflected yet.
(If we use this solution, I will fix it.)

Reply:
> you can acquire the lock faster than 100ms

That’s really convincing 😅. Only need to figure out what to do with dangling , not in demand unlock events you store in the cache…

probably time to look closer into that Redisson solution with pub-sub.

I’m not convinced yet that it is worth to manage one more optional dependency just for this lock solution… although they may have other useful features to consider in the future 😉